app/services/articles/feed.rb

```diff
@@ -0,0 +1,43 @@
+module Articles
+  class Feed
+    attr_reader :stories
+
+    def initialize(number_of_articles:, page:, tag: nil)
+      @number_of_articles = number_of_articles
+      @page = page
+      @tag = tag
+      @stories = published_articles_by_tag
+    end
+
+    def published_articles_by_tag
+      articles = Article.published.limited_column_select.page(@page).per(@num_articles)
+      articles = articles.cached_tagged_with(@tag) if @tag.present? # More efficient than tagged_with
+      articles
+    end
+
+    def time_based_feed(timeframe:)
+      stories.where("published_at > ?", Timeframer.new(timeframe).datetime).
+        order("score DESC")
+    end
+
+    def latest_feed
+      stories.order("published_at DESC").
+        where("featured_number > ? AND score > ?", 1_449_999_999, -40)
+    end
+
+    def default_home_feed(user_signed_in: false)
+      hot_stories = stories.
+        where("score > ? OR featured = ?", 9, true).
+        order("hotness_score DESC")
+      featured_story = hot_stories.where.not(main_image: nil).first
+      if user_signed_in
+        offset = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 2, 2, 2, 3, 3, 4, 5, 6, 7, 8, 9, 10, 11].sample # random offset, weighted more towards zero
+        hot_stories = hot_stories.offset(offset)
+      end
+      new_stories = Article.published.
```

Comment:
We don't need `new_stores` for when `user_signed_in` because we only feature these for signed in users eh?

Before this logic was contained in a context which was wrapped in this conditional in the view.

Reply:
I don't quite follow if that's a question or a comment @benhalpern: are you saying new stories should only be loaded for signed in users? I do think I missed bringing that bit over from `app/views/stories/_main_stories_feed.html.erb`